### PR TITLE
dolphin-emu-beta: add update script, 5.0-15260 -> 5.0-15445

### PR DESCRIPTION
**Pull request by leo60228**
###### Motivation for this change
Some nice features have been added, especially Riivolution support: https://dolphin-emu.org/blog/2021/11/13/dolphin-progress-report-september-and-october-2021/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ashkitten** [comment]
can you use the latest beta version, 5.0-15445?

**Comment by leo60228** [comment]
I would've if not for https://github.com/dolphin-emu/dolphin/pull/10233, but that isn't a massive issue. However, it doesn't seem like this has been consistently done in the past?

**Comment by leo60228** [comment]
I'm not sure if you noticed (just ignore this if you did), but I did switch to the beta version.

**Comment by leo60228** [comment]
I've also added an update script.

**Comment by ashkitten** [comment]
ah, thank you. i tested this yesterday, but forgot to approve changes.

**Review by ashkitten — APPROVED** [review]

File: pkgs/misc/emulators/dolphin-emu/master.nix [hunk 1]
```diff
@@ -64,6 +64,17 @@ stdenv.mkDerivation rec {
...
+
+  passthru.updateScript = writeScript "dolphin-update-script" ''
+  #!/usr/bin/env nix-shell
+  #!nix-shell -i bash -p curl jq common-updater-scripts
+  set -eou pipefail
+  json="$(curl -s https://dolphin-emu.org/update/latest/beta)"
+  version="$(jq -r '.shortrev' <<< "$json")"
+  rev="$(jq -r '.hash' <<< "$json")"
+  update-source-version dolphin-emu-beta "$version" --rev="$rev"
+  '';
```
Comment (thiagokokada) [inline pkgs/misc/emulators/dolphin-emu/master.nix:76]: Can you ident this?

Comment (leo60228) [reply pkgs/misc/emulators/dolphin-emu/master.nix]: Done.

**Comment by leo60228** [comment]
I was completely coincidentally looking at the labels at the exact same time and got confused...

**Review by ashkitten — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]
Result of `nixpkgs-review pr 146933` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dolphin-emu-beta</li>
  </ul>
</details>

Tested with Super Smash Bros. Meele, works fine.

**Pull request merged**